langchain4j-core/src/main/java/dev/langchain4j/internal/ExceptionMapper.java

```diff
@@ -0,0 +1,29 @@
+package dev.langchain4j.internal;
+
+import java.util.concurrent.Callable;
+
+public interface ExceptionMapper extends InvocationPolicy {
+
+    ExceptionMapper DEFAULT = new DefaultExceptionMapper();
+
+    @Override
+    default Callable<?> apply(final Callable<?> action) {
+        return () -> {
+            try {
+                return action.call();
+            } catch (Exception e) {
+                throw mapException(e);
+            }
+        };
+    }
+
+    RuntimeException mapException(Exception e);
+
+    class DefaultExceptionMapper implements ExceptionMapper {
+        @Override
+        public RuntimeException mapException(Exception e) {
+            System.out.println(e.getMessage());
+            return e instanceof RuntimeException re ? re : new RuntimeException(e);
```

Comment:
Default exception mapping implementation should go there. Of course we could implement a different mapping for each language model.